### PR TITLE
SDK Speedup via caching

### PR DESCRIPTION
**Pull request by caesar0x**
**Description**
The SDK was slow AF, this implements caching & speeds things up a decent bit.
Does so by parallelizing few serial calls & caching frequently used / rarely changing ops. Rn cache is timer based w/ relatively fast expiry but could prob be safely extended in number of cases. Also adds (partial) initial cache hydration to speed up first quote. another ape will prob have 2 go through & fix ts/prettier issues.

**Additional context**
seperates out SDK changes  into diff commit 

**Metadata**
- Fixes #[Link to Issue]

## Summary by CodeRabbit

- **New Features**
	- Introduced caching mechanisms across various router components to enhance performance by avoiding redundant computations.
	- Added support for constructor hydration for specific blockchain networks, improving initialization efficiency.

- **Enhancements**
	- Optimized `getBridgeFees` method for concurrent operations, enhancing speed and reliability.

- **New Tools**
	- Launched new utilities for caching (`SimpleCache` and `RouterCache`) and debugging (`logPromise`), facilitating easier development and troubleshooting.

### CODE REVIEW

**Comment by abtestingalpha** [comment]
Not seeing gas drop come through:

Production (w/ ETH gas drop on Arbitrum):
Screen Shot 2024-02-19 at 5 35 55 PM

PR:
Screen Shot 2024-02-19 at 5 36 48 PM

Screen Shot 2024-02-19 at 5 36 37 PM

Separately, seeing repeated calls on hydration (first load logging some 800 instances)
Screen Shot 2024-02-19 at 5 42 36 PM

**Comment by caesar0x** [comment]
ah i see issue.  monke fix

**Comment by caesar0x** [comment]
@abtestingalpha pushed a fix that just turns off the initial hydration (will fix/rewrite initial hydration today) but the rest of the cache seems to b working now

**Comment by caesar0x** [comment]
@abtestingalpha  check out the new version that uses node cache and lmk if you see same issue

**Comment by caesar0x** [comment]
@abtestingalpha, try now, should be fixed.   lmk if this isnt the case

File: packages/sdk-router/src/utils/RouterCache.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+// @ts-nocheck
+/* @ts-ignore */
```
[the remark this answers is not in the document]
Comment (caesar0x) [reply packages/sdk-router/src/utils/RouterCache.ts:2]: how about no

**Comment by caesar0x** [comment]
> Not seeing gas drop come through:
> 
> Production (w/ ETH gas drop on Arbitrum): Screen Shot 2024-02-19 at 5 35 55 PM
> 
> PR: Screen Shot 2024-02-19 at 5 36 48 PM
> 
> Screen Shot 2024-02-19 at 5 36 37 PM
> Separately, seeing repeated calls on hydration (first load logging some 800 instances) Screen Shot 2024-02-19 at 5 42 36 PM

@abtestingalpha, try now, should be fixed. lmk if this isnt the case

**Pull request merged**